internal/runtime/hcsv2/workload_container.go

```diff
@@ -16,6 +18,40 @@ func getWorkloadRootDir(sbid, id string) string {
 	return filepath.Join(getSandboxRootDir(sbid), id)
 }
 
+func createSandboxMountPath(sandboxSource string) error {
+	parentPath, _ := filepath.Split(sandboxSource)
+	pathExt := filepath.Ext(sandboxSource)
+
+	if err := os.MkdirAll(parentPath, 0755); err != nil {
+		return err
+	}
+
+	if pathExt == "" {
+		return os.Mkdir(sandboxSource, 0755)
+	}
+	_, err := os.Create(sandboxSource)
+	return err
+}
+
+func updateSandboxMounts(sbid string, spec *oci.Spec) error {
+	sandboxMountPrefix := "sandbox://"
+	for i, m := range spec.Mounts {
+		if strings.HasPrefix(m.Source, sandboxMountPrefix) {
+			subPath := strings.TrimPrefix(m.Source, sandboxMountPrefix)
+			sandboxSource := filepath.Join(getSandboxMountsDir(sbid), subPath)
```

Comment:
We need to verify that sandboxSource is within `getSandboxMountsDir` after this is joined.

Reply:
What is the concern here?

Reply:
I'm not sure what you mean. We need to verify that the resulting `sandboxSource` is still under `getSandboxMountsDir(id)` after its joined. Consider a path like `..\..\tmp` or something. We dont want people to be able to do that.

Reply:
Makes sense

Reply:
Done